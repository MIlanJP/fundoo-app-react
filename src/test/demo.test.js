import React from 'react'
import SignIn from '../pages/signin'
import{ BrowserRouter,Route} from 'react-router-dom'

// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { render,fireEvent,screen,cleanup } from './testRender'
// import {}  from '@testing-library/react'

afterEach(cleanup)

it('Renders the connected app with initialState', () => {
 const {asFragment,getByTestId}= render(<BrowserRouter><SignIn /></BrowserRouter>, { initialState: { user: 'Redux User' } })
 expect(asFragment()).toMatchSnapshot();
//  expect(getByTestId('login-component'))
 expect(getByTestId('login-label')).toHaveTextContent("Continue to Fundoo")
  expect(screen.getByText(/Create Account/i)).toBeInTheDocument()
})