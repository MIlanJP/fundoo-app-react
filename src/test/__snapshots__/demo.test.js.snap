// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the connected app with initialState 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root mainLogo MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="login-component"
    justify="center"
  >
    <div
      class="MuiCardContent-root"
    >
      <div
        class="LoginLogo"
      >
        <h5
          class="MuiTypography-root Logo MuiTypography-h5"
        >
          <span
            class="f-red"
          >
            F
          </span>
          <span
            class="u-blue"
          >
            u
          </span>
          <span
            class="n-yellow"
          >
            n
          </span>
          <span
            class="d-red"
          >
            d
          </span>
          <span
            class="o-green"
          >
            o
          </span>
          <span
            class="o-blue"
          >
            o
          </span>
        </h5>
      </div>
      <h5
        class="MuiTypography-root signInLabel MuiTypography-h5"
        m="3"
      >
        Sign in
      </h5>
      <p
        class="MuiTypography-root loginInfo MuiTypography-body1"
        data-testid="login-label"
      >
        Continue to Fundoo
      </p>
      <p
        class="message"
      />
      <form
        action=""
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            class="MuiFormControl-root MuiTextField-root EmailInput"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Email Id *
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="outlined-basic"
                name="emailId"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Email Id *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root PasswordInput MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Password *
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="outlined-basic"
                name="password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root showPaswordButtonForSignin"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text LoginButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Log In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
      <a
        class="ForgotPasswordLink"
        href="/forgotpassword"
      >
        Forgot Password?
      </a>
      <a
        class="CreateAccountLink"
        href="/signup"
      >
        Create Account
      </a>
    </div>
  </div>
</DocumentFragment>
`;
